### PR TITLE
Python: ObjectAPI to ValueAPI: UnguardedNextInGenerator

### PR DESCRIPTION
**Pull request by BekaValentine**
This PR is part of the Exceptions modernization effort.

### CODE REVIEW

**Comment by BekaValentine** [comment]
> I'm also curious, how would it handle using a default for `next`, such as `next(iter([]), 'foo')`?

@RasmusWL I suppose it would still flag an error, but this might be undesired. Should we add a new test case to remove false positives?

**Comment by RasmusWL** [comment]
> @RasmusWL I suppose it would still flag an error, but this might be undesired. Should we add a new test case to remove false positives?

Yes. I think we should at least have a test-case for this. (don't know whether we care to handle it with our current security focus).

File: python/ql/test/query-tests/Exceptions/generators/test.py [hunk 1]
```diff
@@ -47,3 +47,6 @@ def next_in_comp(seq, fields):
     seq_iter = iter(seq)
     values = [ next(seq_iter) if f.attname in NAMES else DEFAULT for f in fields ]
     return values
+
+def ok5(seq):
+    yield next(iter([]), 'foo')
```
Comment (RasmusWL) [inline python/ql/test/query-tests/Exceptions/generators/test.py]:
```suggestion
def ok5(seq):
    yield next(iter([]), 'foo')

def ok6(seq):
    yield next(iter([]), default='foo')    
```

File: python/ql/src/Exceptions/UnguardedNextInGenerator.ql [hunk 2]
```diff
@@ -28,6 +28,10 @@ predicate call_to_next(CallNode call, ControlFlowNode iter) {
     iter = next().getArgumentForCall(call, 0)
 }
 
+predicate call_to_next_has_default(CallNode call) {
+    exists(call.getArg(1))
```
Comment (RasmusWL) [inline python/ql/src/Exceptions/UnguardedNextInGenerator.ql]:
```suggestion
    exists(call.getArg(1)) or exists(call.getArgByName("default"))
```

**Pull request merged**